fitting/bias_test.py

```diff
@@ -0,0 +1,504 @@
+
+
+def bias_test(options):
+
+    label = options.label
+    plot_dir = options.plotDir
+    if(plot_dir[-1] != '/'):
+        plot_dir += '/'
+
+    if(not os.path.exists(plot_dir)):
+        os.system("mkdir %s" % plot_dir)
+
+    fout_name = plot_dir + "bias_test_results_%s_%s" % (options.mass, label)
+    if(os.path.exists(fout_name + ".json")):
+        #remove old results
+        os.system("rm %s" % fout_name + ".json")
+
+    fine_bin_size = 4
+    mass = options.mass 
+
+    binsx = [1460, 1530, 1607, 1687, 1770, 1856, 1945, 2037, 2132, 2231, 2332, 2438,
+             2546, 2659, 2775, 2895, 3019, 3147, 3279, 3416, 3558, 3704, 3854,
+             4010, 4171, 4337, 4509, 4700, 4900,  5100, 5300, 5500, 5800,
+             6100, 6400, 6800]
+
+    if(options.mjj_max < 0. and options.rebin): 
+        options.mjj_max = get_mjj_max(options.inputFile) + 5.0
+        options.mjj_max = max(1.2*options.mass, options.mjj_max)
+    #print("MJJ MAX %.2f" % options.mjj_max)
+    
+    if(options.mjj_min > 0 and options.mjj_min < binsx[-1]):
+        start_idx = 0
+        while(binsx[start_idx] < options.mjj_min):
+            start_idx +=1
+        binsx = binsx[start_idx:]
+
+        if(abs(options.mjj_min - binsx[0]) < 50.):
+            binsx[0] = options.mjj_min
+        else:
+            binsx.insert(0, options.mjj_min)
+        print("Will start fit from %.0f GeV" % binsx[0])
+
+    if(options.mjj_max > 0 and options.mjj_max < binsx[-1]):
+        print("rebinning with max mjj %.2f" % options.mjj_max)
+        end_idx = len(binsx)-1
+        while(binsx[end_idx]   > options.mjj_max and end_idx > 0): 
+            end_idx -=1
+        binsx = binsx[:end_idx]
+
+        if(abs(options.mjj_max - binsx[-1]) < 50.):
+            binsx[-1] = options.mjj_max
+        else:
+            binsx.append(options.mjj_max)
+        print("Will end fit at %.0f GeV" % binsx[-1])
+        print(binsx)
+
+
+    # round to smallest precision we are storing mass values with, otherwise
+    # get weird effects related to bin size
+    roundTo(binsx, fine_bin_size)
+
+
+
+    nbins_fine = int(binsx[-1] - binsx[0])/fine_bin_size
+
+    histos_sb = ROOT.TH1F("mjj_sb", "mjj_sb" ,nbins_fine, binsx[0], binsx[-1])
+    
+    
+    load_h5_sb(options.inputFile, histos_sb)
+    print("************ Found %i total events \n" % histos_sb.GetEntries())
+    num_evts = histos_sb.GetEntries()
+
+    
+
+    if(options.rebin):
+        bins_nonzero = get_rebinning(binsx, histos_sb)
+        print("Rebinning to avoid zero bins!")
+        print("old", binsx)
+        print("new", bins_nonzero)
+        bins = bins_nonzero
+    else:
+        bins = binsx
+    roobins = ROOT.RooBinning(len(bins)-1, array('d', bins), "mjjbins")
+
+    if(options.refit_sig):
+        print ("########## FIT SIGNAL AND SAVE PARAMETERS ############")
+        sig_file_name = "sig_fit.root"
+
+        fit_signalmodel(options.inputFile, sig_file_name, mass, binsx, nbins_fine, plot_dir,return_fit=False,
```

Comment:
Ah yeah. It probably doesn't make sense to fit the signal from the input file anyways for a bias test because the input should be background only. I'll remove the command line option for this